app/components/post_list/post/body/content/permalink_preview/index.tsx

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {withDatabase, withObservables} from '@nozbe/watermelondb/react';
+import {of as of$} from 'rxjs';
+import {switchMap, distinctUntilChanged} from 'rxjs/operators';
+
+import {fetchUsersByIds} from '@actions/remote/user';
+import {withServerUrl} from '@context/server';
+import {observePost} from '@queries/servers/post';
+import {observeConfigBooleanValue} from '@queries/servers/system';
+import {observeUserOrFetch, observeTeammateNameDisplay, observeCurrentUser} from '@queries/servers/user';
+
+import PermalinkPreview from './permalink_preview';
+
+import type {WithDatabaseArgs} from '@typings/database/database';
+
+const enhance = withObservables(['embedData', 'serverUrl'], ({database, embedData, serverUrl}: WithDatabaseArgs & {embedData: PermalinkEmbedData; serverUrl?: string}) => {
+    const showPermalinkPreviews = observeConfigBooleanValue(database, 'EnablePermalinkPreviews', false);
+    const teammateNameDisplay = observeTeammateNameDisplay(database);
+
+    const userId = embedData?.post?.user_id;
+    const author = userId ? observeUserOrFetch(database, serverUrl || '', userId, fetchUsersByIds) : of$(undefined);
+
+    const currentUser = observeCurrentUser(database);
+    const locale = currentUser.pipe(
+        switchMap((u) => of$(u?.locale || 'en')),
+        distinctUntilChanged(),
+    );
```

Comment:
you can replace this with the context found here https://github.com/mattermost/mattermost-mobile/blob/main/app/context/user_locale/index.tsx which can give you a hook to get the locale